### DISCUSSION TITLE
A Little Question about State Change

### DISCUSSION THREAD

**Discussion by ywleeee** [pmndrs/zustand#1525] [Q&A] [answered]
Hi everyone 
When the state changes, zustand will tell all components that have `useStore` in them to rerender, is there any way to know when all the rerenders are done? Because in my case, I need to run a function right after all the changes is done
I'm new to zustand, thanks in advance

**Comment by dai-shi** [comment, marked as answer]
In terms of finishing browser painting, `useEffect` is the right solution.
However, it's often unnecessary, and you can assume, by updating a store, it's done.

**Reply by ywleeee** [reply]
Thank you so much!!
